rfcs/text/0011_global_search.md

```diff
@@ -0,0 +1,693 @@
+be used when it is not technically possible to register it from the server side instead.
+
+### results url consolidation
+
+When retrieving results from providers, the GS service will always start by consolidating them. The most notable (and
+currently only) step is to convert the result url to an absolute one.
+
+#### absolute url conversion logic
+
+Results returned from the GlobalSearch's `find` programmatic and HTTP APIs will all contains absolute urls for the following
+reasons:
+- It would not be usable by external consumers otherwise. 
+- Some result providers are supposed to be returning results from outside of the current `basePath`, or even from outside of the
+  Kibana instance (ie: results from another space, results from another Kibana instance in a cloud cluster)
+
+However, as forging absolute urls can be a tedious process for the plugins, the `url` property of results returned by
+a result provider can be either an absolute url or an url path relative to the executing request's `basePath`.
+
+I.E are considered valid:
+- `https://my-other-kibana-instance/some/result`
+- `/app/kibana#/dashboard/some-id`
+
+When consolidating the result, the logic regarding the `url` property is:
+- if the `url` is absolute, return it unchanged
+- if the `url` is a relative path (starts with `/`), it will be converted to an absolute url by prepending the Kibana
+instance's newly introduced `publicAddress`.
+
+#### server.publicAddress
+
+Given the fact that some Kibana deployments have complex architecture (proxies, rewrite rules...), there is currently
+no reliable way to know for sure what the public address used to access kibana is (at least from the server-side).
+
+A new `server.publicAddress` property will be added to the kibana configuration, allowing ops to explicitly define the public
+address to instance is accessible from. This property is only meant to be used to generate 'links' to arbitrary resources
+and will not be used to configure the http server in any way.
+
+When not explicitly defined, this property will be computed using the known `server` configuration values:
+
+```ts
+const defaultPublicAddress = removeTrailingSlash(
+  `${getServerInfo().protocol}://${httpConfig.host}:${httpConfig.port}/${httpConfig.basePath}`
+);
+
+const getPublicAddress = () => httpConfig.publicAddress ?? defaultPublicAddress;
+```
+
+A new `getAbsoluteUrl` api will also be added to the core `http` service contract:
+
+```ts
+const getAbsoluteUrl = (path: string, request: KibanaRequest) => {
+  const publicUrl = getPublicAddress();
+  const absoluteUrl = joinRemovingDuplicateAndTrailingSlash(
+    publicUrl,
+    // note: this is actually wrong. We would need the `requestScopePath` here
+    // as this currently returns `${this.serverBasePath}${requestScopePath}` and the basePath
+    // is already included in `publicAddress`
+    serverContract.basePath.get(request), 
+    path
```

Comment:
We will need something like `basePath.getRequestScopePath` for that. However the implementation is trivial as the basePath service is still aggregating the basePath with the extra path in `${this.serverBasePath}${requestScopePath}`. @restrry do you see any issue in adding this API?